### DISCUSSION TITLE
Request for Option to Retain Original Video Encoding Settings in Export

### DISCUSSION THREAD

**Discussion by carllx** [mifi/lossless-cut#2023] [Q&A] [answered]
Hello Lossless Cut Community,

I hope this message finds you well. I've been using Lossless Cut and have found it to be an incredibly useful tool for my video editing needs. However, I've encountered a situation where I need to retain the original encoding settings of the video during the export process. Currently, the app executes the following FFmpeg command:

```bash
ffmpeg \
-hide_banner \
-ss '447.62911' \
-i '$Input' \
-t '14.99236' \
-avoid_negative_ts make_zero \
-map '0:v' '-c:0' copy \
-map '0:a' '-c:1' copy \
-map_metadata 0 \
-movflags use_metadata_tags \
-movflags '+faststart' \
-default_mode infer_no_subs \
-ignore_unknown \
-strict experimental \
-f mp4 \
-y '$Output'
```
In this command, the `-map '0:v' '-c:0' copy and -map '0:a' '-c:1' copy` options are used to copy the video and audio streams respectively. However, this results in the original video channel's encoding settings being removed, which is not ideal for my use case.

I am wondering if it would be possible to add an option that allows users to either keep or remove these -map settings based on their preferences? This would be extremely helpful for those of us who need to preserve the original encoding quality and settings of the video.

**Comment by mifi** [comment]
Hi! I'm not sure exactly what you're asking for. you want to remove `-map 0:v` etc? why do you want to remove it? if you go to the Tracks panel in the UI and disable the video track, then `-map 0:v` will go away.

**Reply by carllx** [reply]
Thank you for your patience and reply. Perhaps my inquiry was not expressed clearly enough, so I will try to explain my specific needs in detail.

My primary aim is to retain the original encoding settings of the video file when exporting videos using Lossless Cut. By encoding settings, I am referring to parameters such as cabac, crf, ref frames, B-frames, etc., which are known as encoding settings in FFmpeg.

The default FFmpeg command used by Lossless Cut is as follows:

```shell
ffmpeg -hide_banner -ss '447.62911' -i '$Input' -t '14.99236' -avoid_negative_ts make_zero -map '0:v' '-c:0' copy -map '0:a' '-c:1' copy -map_metadata 0 -movflags use_metadata_tags -movflags '+faststart' -default_mode infer_no_subs -ignore_unknown -strict experimental -f mp4 -y '$Output'
```
This directly copies the video and audio streams. However, the result is that the original encoding settings information is removed from the output file. For example, if the original video has specific encoding settings (such as crf=18.0, cabac=1 or ref=5), I would like these settings to remain in the video file processed by Lossless Cut.
```
(Extract from MediaInfo)
Encoding settings: cabac=1 / ref=5 / deblock=1:0:0 / analyse=0x3:0x113 / me=hex / subme=8 / psy=1 / psy_rd=1.00:0.00 / mixed_ref=1 / me_range=16 / chroma_me=1 / trellis=2 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=-2 / threads=8 / lookahead_threads=1 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / bframes=3 / b_pyramid=2 / b_adapt=1 / b_bias=0 / direct=3 / weightb=1 / open_gop=0 / weightp=2 / keyint=240 / keyint_min=121 / scenecut=0 / intra_refresh=0 / rc_lookahead=50 / rc=crf / mbtree=1 / crf=18.0 / qcomp=0.60 / qpmin=10 / qpmax=69 / qpstep=4 / vbv_maxrate=32000 / vbv_bufsize=32000 / crf_max=0.0 / nal_hrd=none / filler=0 / ip_ratio=1.41 / aq=1:1.00
```
Upon conducting some experiments, I found that when I remove the `-map '0:v' '-c:0' copy` and `-map '0:a' '-c:1' copy` options from the FFmpeg command, the output file retains some of the original encoding information. This has led me to contemplate whether it would be possible to adjust the FFmpeg command or parameters to meet my needs.

Therefore, my question is, would it be possible to add a feature or option within Lossless Cut that allows users to customize FFmpeg's transcoding parameters? More specifically, could users be given the choice to retain or remove the `-map '0:v' '-c:0' copy` and `-map '0:a' '-c:1' copy` options? Such a feature would be invaluable to users who need precise control over the encoding of their output videos.

**Reply by mifi** [reply, marked as answer]
Correct me if I'm wrong, but I believe that if you remove `-map '0:v' '-c:0' copy` and `-map '0:a' '-c:1' copy` from the command, then ffmpeg will no longer do a lossless copy. it will instead re-encode all streams which is slow and leads to quality loss. That's out of the scope of losslesscut, as its main purpose is lossless operations.
